docs/tasks/stateful-sets/deleting-pods.md

```diff
@@ -0,0 +1,68 @@
+
+### Deletion of Pods
+
+#### Graceful Deletion
+
+Graceful deletion may be performed using the following:
+
+```shell
+$ kubectl delete pods <pod>
+```
+
+For the above to lead to graceful termination, the pod **must** not specify a `pod.Spec.TerminationGracePeriodSeconds` of 0. The practice of setting a `pod.Spec.TerminationGracePeriodSeconds` of 0 seconds is unsafe and strongly discouraged for Stateful Set pods. Graceful deletion is safe and will ensure that the pod shuts down gracefully before the kubelet deletes the name from the apiserver (*TODO: link to http://kubernetes.io/docs/user-guide/pods/#termination-of-pods). 
+
+ In order for Stateful Sets to provide *at most one* semantics, changes were made to the Node Controller in 1.5, (*TODO: Link to Node Controller behavior page.*) so that it never force deletes pods on unreachable nodes. When a node becomes unreachable for an extended period of time, we deem such a node as having been partitioned. The pods running on an unreachable node enter the 'Terminating' or 'Unknown' state eventually. Pods may also enter these states when the user attempts graceful deletion (*link to kubectl documentation*) of a pod on a partitioned node. The only ways in which a pod in such a state can be removed from the apiserver are as follows:
+   * The node object is deleted (either by the user, or by the node controller).
+   * The kubelet on the unresponsive node starts responding, kills the pod and removes the entry from the apiserver. 
+   * Force deletion of the pod by the user.
+   
+As explained above, there are cases in which graceful deletion will be unable to remove a pod. In these cases, if you want the Stateful Set controller to create a replacement pod with the same identity, you may want to perform force deletion.
+
+#### Force Deletion
+
+Force deletions do **not** wait for confirmation from the kubelet that the pod has been terminated. Irrespective of whether a force deletion is successful in killing a pod, it will immediately free up the name from the apiserver. This would let the Stateful Set controller create a replacement pod with that same identity. However, the risks of doing so must be evident. If there is a pod running on a node that has been partitioned from the apiserver, it may still be able to communicate with other pods in the Stateful Set. Force deletion of a Stateful Set pod is a user's assertion that the pod in question is "fenced" and its name can be safely freed up for a replacement to be created. 
```

Comment:
Elsewhere it seems like we're trying to avoid assuming users know what "fenced" means. Maybe elaborate a bit: "... is a user's assertion that the pod in question will never again make contact with other pods in the Set."